multipatch_analysis/pipeline/morphology.py

```diff
@@ -65,20 +116,57 @@ def ready_jobs(self):
         """
         # All experiments and their creation times in the DB
         expts = ExperimentPipelineModule.finished_jobs()
-        
+
         # Look up nwb file locations for all experiments
         session = db.Session()
-        expt_recs = session.query(db.Experiment.acq_timestamp, db.Experiment.storage_path).all()
-        expt_paths = {rec.acq_timestamp: rec for rec in expt_recs}
+        # expts = session.query(db.Experiment).filter(db.Experiment.acq_timestamp==1521667891.153).all()
         session.rollback()
-        
+        morpho_results = morpho_db()
         # Return the greater of NWB mod time and experiment DB record mtime
         ready = OrderedDict()
         for expt_id, (expt_mtime, success) in expts.items():
             if success is not True:
                 continue
-            rec = expt_paths[expt_id]
-            pip_file = os.path.join(config.synphys_data, rec.storage_path, 'pipettes.yml')
-            pip_mtime = timestamp_to_datetime(os.stat(pip_file).st_mtime)
-            ready[rec.acq_timestamp] = max(expt_mtime, pip_mtime)
+
+            expt = session.query(db.Experiment).filter(db.Experiment.acq_timestamp==expt_id).all()[0]
+            cluster_cells = get_expt_cell_cluster(expt)
+            if cluster_cells is None:
+                continue
+            cell_hash_compare = []
+            for cell in cluster_cells:
+                if cell.id is None:
+                    continue
+                morpho_db_hash = hash(';'.join(filter(None, morpho_results.get(cell.id, [None]))))
+                cell_morpho_rec = session.query(db.Morphology).filter(db.Morphology.cell_specimen_id==str(cell.id)).all()
+                if len(cell_morpho_rec) == 1:
+                    cell_rec_hash = cell_morpho_rec[0].morpho_db_hash
+                    cell_hash_compare.append(morpho_db_hash == cell_rec_hash)
+                else:
+                    cell_hash_compare.append(False)
+            if all(cell_hash_compare) is False:
+                ready[expt.acq_timestamp] = datetime.datetime.now()
+    
         return ready
+
+def morpho_db():
+    cnxn_str = r'DRIVER={Microsoft Access Driver (*.mdb, *.accdb)}; DBQ=%s' % config.morpho_address
+    cnxn = pyodbc.connect(cnxn_str)
+    cursor = cnxn.cursor()
+    morpho_table = cursor.execute('select * from MPATCH_CellsofCluster')
+    results = morpho_table.fetchall()
+    morpho_results = {int(r.cell_specimen_id): r for r in results}
+
+    return morpho_results
+
+def get_expt_cell_cluster(expt):
```

Comment:
Seems like something that could be in `lims.py` or a method of `Experiment` ?

Reply:
or stored in the Experiment table as it's supposed to be :)